icssc/ZotMeet
# Add authentication

![2024-12-04 00 51 26](https://github.com/user-attachments/assets/1e55d55c-d014-43ea-a189-ed9878669176)

## Summary
- Added utilities for auth outlined from [lucia](https://lucia-auth.com/sessions/basic-api/drizzle-orm)
  - basic sessions
  - cookies with Next.JS
  - password hashing and verification with `@node-rs/argon2`
- Added shadcn dialog modal for sign up and login (button may change, might be worth asking UI/UX team)
- Added shadcn form with interactive state & zod schema for sign up and login
- Added server actions for sign up and login
- Added server action for sign out and modified sidebar to dynamically change based on user state

### Important APIs added
- `src/lib/auth/index.ts`: added `getCurrentSession` function
  - returns a Promise of the current user and session.
  - If no user is logged in, returns null when awaited

### Testing
- Checkout this branch
- Install dependencies
- Run migrations + dev server

- Test signup modal
  - invalid email
  - empty password
  - sign up new user, should be automatically logged in
- Refresh page to see user persists
  - check browser console > applications > cookies to see session cookie's expiration date to be 30 days from now
- Sign out 
- Test sign in modal
  - empty password
  - invalid email
  - incorrect password
 
Closes #130
